modules/mini-games/Race.lua

```diff
@@ -252,177 +279,177 @@ local function insideBox(box, pos)
-
-local token_for_kill_biters = Token.register(kill_biters)
+--- Make the car and player destructible again
+local stop_invincibility = Token.register(function(name)
+    variables["Dead_car"][name].car.destructible = true
+    local player = variables["Dead_car"][name].player
+    player.character.destructible = true
+end)
 
+--- Kill all biters in a close range to the player
+local kill_biters = Token.register(function(name)
+    local biters = surface[1].find_enemy_units(variables["Dead_car"][name].position, 3, "player")
+    for i, biter in ipairs(biters) do
+        biter.destroy()
+    end
+end)
 
-local respawn_car = function(name)
+--- Respawn the car for a player
+local respawn_car = Token.register(function(name)
     local player = variables["Dead_car"][name].player
-    local car =
-        surface[1].create_entity {
+    local position = surface[1].find_non_colliding_position('car', variables["Dead_car"][name].position, 5, 0.5)
+    position = position or variables["Dead_car"][name].position
+
+    local car = surface[1].create_entity {
         name = "car",
         direction = defines.direction.north,
-        position = variables["Dead_car"][name].position,
+        position = position,
         force = "player"
     }
+
+    car.operable = false
     car.set_driver(player)
     car.orientation = variables["Dead_car"][name].orientation
-    car.get_fuel_inventory().insert({name = variables["fuel"], count = 100})
+    car.get_fuel_inventory().insert{ name = variables["fuel"], count = 100 }
     cars[name] = car
 
     Permission_Groups.set_player_group(player, variables["Dead_car"][name].group)
-    invisabilty(car,name)
+    start_invincibility(car, name)
 
     variables["Dead_car"][name].car = car
-end
-
-token_for_car = Token.register(respawn_car)
-
+end)
 
+--- Triggered when an entity is destroyed, used to respawn the car
 local car_destroyed = function(event)
     local dead_car = event.entity
-    if dead_car.name == "car" then
-        local player = dead_car.get_driver().player
-        local name = player.name
-        if not variables["Dead_car"] then
-            variables["Dead_car"] = {}
-        end
-        if not variables["Dead_car"][name] then
-            variables["Dead_car"][name] = {}
-        end
-        variables["Dead_car"][name].player = player
-        variables["Dead_car"][name].position = dead_car.position
-        variables["Dead_car"][name].orientation = dead_car.orientation
-        variables["Dead_car"][name].group = Permission_Groups.get_group_from_player(player).name
-        player.character.destructible = false
-        Permission_Groups.set_player_group(player, "out_car")
-        task.set_timeout_in_ticks(180, token_for_car,name)
-        task.set_timeout_in_ticks(190, token_for_kill_biters,name)
-        task.set_timeout_in_ticks(480, token_for_stop_invins,name)
+    if dead_car.name ~= "car" then return end
+    if not variables["Dead_car"] then
```

Comment:
variables.Dead_car could be set in start so it does not have tho check every time

Reply:
Yes I agree I will add dead_car to the variables list and have it reset on game stop.